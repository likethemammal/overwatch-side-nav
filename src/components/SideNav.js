import React, { Component } from 'react'

import css from './SideNav.less'

import Item from './Item'
import Scroll from './Scroll'

import {
    ids as _ids,
} from '../selectors'

export default class SideNav extends Component {
    static defaultProps = {
        className: css.container,
        zIndex: 1000000,
    }

    state = {
        windowHash: window.location.hash,
    }

    onHashChange = (windowHash) => {
        this.setState({
            windowHash,
        })
    }

    render() {

        const { className, items, zIndex, ...rest } = this.props
        const { windowHash } = this.state
        const shared_state = {
            windowHash,
            onHashChange: this.onHashChange,
        }
        const containerStyles = {
            zIndex,
        }

        return <div
            className={className}
            style={containerStyles}
        >
            <Scroll
                {...{items}}
                {...shared_state}
            />
            {items.map((item, i) => <Item
                key={i}
                {...shared_state}
                {...rest}
                {...item}
            />)}
        </div>
    }
}