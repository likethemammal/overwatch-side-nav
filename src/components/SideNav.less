@import "../colors.less";

.container {
    position: fixed;
    right: 0;
    color: white;
    top: 50%;
    transform: translateY(-50%);
    z-index: 1;
}