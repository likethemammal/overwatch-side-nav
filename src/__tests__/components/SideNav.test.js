import React from 'react';
import {shallow} from 'enzyme';

import general from '@likethemammal/general-components'

const shared = general.__tests__._shared.tests

import SideNav from '../../'

const items = [
    {
        label: 'Label',
        id: 'some_id',
        icon: [],
        Component: <div>test</div>,
    },
    {
        label: 'Test',
        id: 'slug',
        icon: [],
        Component: <div>The fuck</div>,
    },
]


describe('SideNav', () => {

    const component = shallow(
        <SideNav
            items={items}
        />
    )

    shared.SNAPSHOTS_SHOULD_MATCH([
        component
    ])

})