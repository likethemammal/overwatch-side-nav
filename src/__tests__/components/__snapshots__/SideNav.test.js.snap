// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Snapshot 0 should match 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideNav
    className={undefined}
    items={
      Array [
        Object {
          "icon": "svg-file-stub",
          "id": "some_id",
          "label": "Label",
        },
        Object {
          "icon": "svg-file-stub",
          "id": "slug",
          "label": "Test",
        },
      ]
    }
    zIndex={1000000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Scroll
          items={
            Array [
              Object {
                "icon": "svg-file-stub",
                "id": "some_id",
                "label": "Label",
              },
              Object {
                "icon": "svg-file-stub",
                "id": "slug",
                "label": "Test",
              },
            ]
          }
          onHashChange={[Function]}
          windowHash=""
        />,
        Array [
          <Item
            activeColor="#00c3ff"
            className={undefined}
            defaultColor="#323c50"
            icon="svg-file-stub"
            id="some_id"
            label="Label"
            onHashChange={[Function]}
            textColor="white"
            windowHash=""
          />,
          <Item
            activeColor="#00c3ff"
            className={undefined}
            defaultColor="#323c50"
            icon="svg-file-stub"
            id="slug"
            label="Test"
            onHashChange={[Function]}
            textColor="white"
            windowHash=""
          />,
        ],
      ],
      "className": undefined,
      "style": Object {
        "zIndex": 1000000,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "items": Array [
            Object {
              "icon": "svg-file-stub",
              "id": "some_id",
              "label": "Label",
            },
            Object {
              "icon": "svg-file-stub",
              "id": "slug",
              "label": "Test",
            },
          ],
          "onHashChange": [Function],
          "windowHash": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "activeColor": "#00c3ff",
          "className": undefined,
          "defaultColor": "#323c50",
          "icon": "svg-file-stub",
          "id": "some_id",
          "label": "Label",
          "onHashChange": [Function],
          "textColor": "white",
          "windowHash": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "activeColor": "#00c3ff",
          "className": undefined,
          "defaultColor": "#323c50",
          "icon": "svg-file-stub",
          "id": "slug",
          "label": "Test",
          "onHashChange": [Function],
          "textColor": "white",
          "windowHash": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Scroll
            items={
              Array [
                Object {
                  "icon": "svg-file-stub",
                  "id": "some_id",
                  "label": "Label",
                },
                Object {
                  "icon": "svg-file-stub",
                  "id": "slug",
                  "label": "Test",
                },
              ]
            }
            onHashChange={[Function]}
            windowHash=""
          />,
          Array [
            <Item
              activeColor="#00c3ff"
              className={undefined}
              defaultColor="#323c50"
              icon="svg-file-stub"
              id="some_id"
              label="Label"
              onHashChange={[Function]}
              textColor="white"
              windowHash=""
            />,
            <Item
              activeColor="#00c3ff"
              className={undefined}
              defaultColor="#323c50"
              icon="svg-file-stub"
              id="slug"
              label="Test"
              onHashChange={[Function]}
              textColor="white"
              windowHash=""
            />,
          ],
        ],
        "className": undefined,
        "style": Object {
          "zIndex": 1000000,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "items": Array [
              Object {
                "icon": "svg-file-stub",
                "id": "some_id",
                "label": "Label",
              },
              Object {
                "icon": "svg-file-stub",
                "id": "slug",
                "label": "Test",
              },
            ],
            "onHashChange": [Function],
            "windowHash": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "activeColor": "#00c3ff",
            "className": undefined,
            "defaultColor": "#323c50",
            "icon": "svg-file-stub",
            "id": "some_id",
            "label": "Label",
            "onHashChange": [Function],
            "textColor": "white",
            "windowHash": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "activeColor": "#00c3ff",
            "className": undefined,
            "defaultColor": "#323c50",
            "icon": "svg-file-stub",
            "id": "slug",
            "label": "Test",
            "onHashChange": [Function],
            "textColor": "white",
            "windowHash": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
